vllm/entrypoints/openai/api_server.py

```diff
@@ -70,50 +62,21 @@ async def check_model(request) -> Optional[JSONResponse]:
 
 
 async def get_gen_prompt(request) -> str:
-    if not _fastchat_available:
-        raise ModuleNotFoundError(
-            "fastchat is not installed. Please install fastchat to use "
-            "the chat completion and conversation APIs: `$ pip install fschat`"
-        )
-    if version.parse(fastchat.__version__) < version.parse("0.2.23"):
-        raise ImportError(
-            f"fastchat version is low. Current version: {fastchat.__version__} "
-            "Please upgrade fastchat to use: `$ pip install -U fschat`")
-
-    conv = get_conversation_template(request.model)
-    conv = Conversation(
-        name=conv.name,
-        system_template=conv.system_template,
-        system_message=conv.system_message,
-        roles=conv.roles,
-        messages=list(conv.messages),  # prevent in-place modification
-        offset=conv.offset,
-        sep_style=SeparatorStyle(conv.sep_style),
-        sep=conv.sep,
-        sep2=conv.sep2,
-        stop_str=conv.stop_str,
-        stop_token_ids=conv.stop_token_ids,
-    )
-
-    if isinstance(request.messages, str):
-        prompt = request.messages
+    if chat_template is not None:
+        return tokenizer.apply_chat_template(
+            conversation=request.messages,
+            chat_template=chat_template,
+            tokenize=False,
+            add_generation_prompt=request.add_generation_prompt)
+    elif tokenizer.chat_template is not None:
```

Comment:
```suggestion
    elif hasattr(tokenizer, 'chat_template') and tokenizer.chat_template is not None:
```

We can probably clean this up and just apply the following

```python
if chat_template is not None:
  tokenizer.chat_template = chat_template

tokenizer.apply_chat_template(...)
```

Reply:
Sure, I had actually done that originally, but then noticed the value in the constructor...will go back to that implementation.